source/fundamentals/gridfs.txt

```diff
@@ -0,0 +1,387 @@
+code example shows you how you can call the ``downloadToStream()`` method to
+download the first revision of the file named "myProject.zip" to an
+``OutputStream``.
+
+.. literalinclude:: /includes/fundamentals/code-snippets/GridFSOperations.java
+   :language: java
+   :dedent:
+   :start-after: start downloadToStream
+   :end-before: end downloadToStream
+
+For more information on this method, see the
+:java-docs:`downloadToStream() <apidocs/mongodb-driver-sync/com/mongodb/client/gridfs/GridFSBucket.html#downloadToStream(java.lang.String,java.io.OutputStream,com.mongodb.client.gridfs.model.GridFSDownloadOptions)>`
+API documentation.
+
+Download a File to an Input Stream
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You can download a file in a GridFS bucket to memory by using an input
+stream. You can call the ``openDownloadStream()`` method on the GridFS
+bucket to open a ``GridFSDownloadStream``, an input stream from which you
+can read the file.
+
+The following code example shows you how to download a file referenced by
+the ``fileId`` variable into memory:
+
+.. literalinclude:: /includes/fundamentals/code-snippets/GridFSOperations.java
+   :language: java
+   :dedent:
+   :start-after: start downloadToMemory
+   :end-before: end downloadToMemory
+
+For more information on this method, see the
+:java-docs:`openDownloadStream()
+<apidocs/mongodb-driver-sync/com/mongodb/client/gridfs/GridFSBucket.html#openDownloadStream(java.lang.String,com.mongodb.client.gridfs.model.GridFSDownloadOptions)>`.
+API documentation.
+
+.. _gridfs-rename-files:
+
+Rename Files
+------------
+
+You can update the name of a GridFS file in your bucket by calling the
+``rename()`` method. You must specify the file to rename by its file id
+rather than its file name.
+
+.. note::
+
+   The ``rename()`` method only supports updating the name of a single file.
+   To rename multiple files, retrieve a list of files matching the file name
+   from the bucket, extract the file id values from the files you want to
+   rename, and pass each file id in separate calls to the ``rename()`` method.
+
+The code example below shows you how to update the name of the file referenced
+by the ``fileId`` variable to "mongodbTutorial.zip":
+
+.. literalinclude:: /includes/fundamentals/code-snippets/GridFSOperations.java
+   :language: java
+   :dedent:
+   :start-after: start renameFile
+   :end-before: end renameFile
+
+For more information on this method, see the
+:java-docs:`rename() <apidocs/mongodb-driver-sync/com/mongodb/client/gridfs/GridFSBucket.html#rename(org.bson.types.ObjectId,java.lang.String)>`
+API documentation.
+
+.. _gridfs-delete-files:
+
+Delete Files
+------------
+
+You can remove a file from your GridFS bucket by calling the ``delete()``
+method. You must specify the file by the value of its file id rather than
+its file name.
+
+.. note::
+
+   The ``delete()`` method only supports deleting one file at a time. To
```

Comment:
Similar as above: rename uses the wording "only supports updating the name of a single file", maybe we should use the same wording here "only supports  deleting a single file" (or vice versa).

Reply:
Updating the rename() note.